tools/watermill-cli/cmd/kafka.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/spf13/viper"
+
+	"github.com/ThreeDotsLabs/watermill/message/infrastructure/kafka"
+	"github.com/spf13/cobra"
+)
+
+var kafkaCmd = &cobra.Command{
+	Use:   "kafka",
+	Short: "Commands for the kafka Pub/Sub provider",
+	Long: `Consume or produce messages from the kafka Pub/Sub provider.
+
+For the configuration of consuming/producing of the messages, check the help of the relevant command.`,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		err := rootCmd.PersistentPreRunE(cmd, args)
+		if err != nil {
+			return err
+		}
+		logger.Debug("Using kafka pub/sub", nil)
+
+		brokers := viper.GetStringSlice("kafka.brokers")
+
+		if cmd.Use == "consume" {
+			saramaSubscriberConfig := kafka.DefaultSaramaSubscriberConfig()
+
+			if viper.GetBool("kafka.consume.fromBeginning") {
+				logger.Trace("Configured sarama to consume messages from beginning", nil)
+				// equivalent of auto.offset.reset: earliest
+				saramaSubscriberConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+			}
+
+			consumer, err = kafka.NewSubscriber(
+				kafka.SubscriberConfig{
+					Brokers:       brokers,
+					ConsumerGroup: viper.GetString("kafka.consume.consumerGroup"),
+				},
+				saramaSubscriberConfig,
+				kafka.DefaultMarshaler{},
+				logger,
+			)
+			if err != nil {
+				return err
+			}
+		}
+
+		if cmd.Use == "produce" {
+			producer, err = kafka.NewPublisher(brokers, kafka.DefaultMarshaler{}, nil, logger)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(kafkaCmd)
+
+	kafkaCmd.PersistentFlags().StringP(
+		"topic",
+		"t",
+		"",
+		"The topic to produce messages to (produce) or consume message from (consume)",
+	)
+	ensure(kafkaCmd.MarkPersistentFlagRequired("topic"))
+	ensure(viper.BindPFlag("kafka.topic", kafkaCmd.PersistentFlags().Lookup("topic")))
+
+	consumeCmd := addConsumeCmd(kafkaCmd, "kafka.topic")
+	_ = addProduceCmd(kafkaCmd, "kafka.topic")
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	kafkaCmd.PersistentFlags().StringSliceP("brokers", "b", nil, "A list of kafka brokers")
+	ensure(kafkaCmd.MarkPersistentFlagRequired("brokers"))
+	ensure(viper.BindPFlag("kafka.brokers", kafkaCmd.PersistentFlags().Lookup("brokers")))
+
+	consumeCmd.PersistentFlags().Bool("fromBeginning", false, "Equivalent to auto.offset.reset: earliest")
```

Comment:
maybe spinal-case? also for another flags?

Reply:
Yeah, spinal should be better (more POSIX or whatever). I thought cobra recommended camelCase, but I must have dreamed that up

Reply:
b3ae9f252ccb6e90efcf9f72512765a9df71cff4 done.
Funnily enough, the tool still works